app/components/quiz/QuizQuestion.js

```diff
@@ -19,20 +19,24 @@ export default React.createClass({
     return {colors: randomColors, isSelectedFlags: isSelected};
   },
 
-  answerClicked: function(id, selected) {
+  answerProcessed: function(id, selected, text=null) {
```

Comment:
You mean call other functions inside this one that handles each type? Or break up answerProcessed itself? I did it like that because answerProcessed is a prop for QuizAnswer.